### PR TITLE
Created a breadcrumb observer that can be injected in order to minimize modifications in core component

### PR DESCRIPTION
**Pull request by bf319**
Created the IBreadcrumbObserver interface which registers the tree from the breadcrumb picker and adds the focus icon and functionality for changing the root.

### CODE REVIEW

File: src/vs/workbench/contrib/scopeTree/browser/breadcrumbObserver.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { Tree } from 'vs/workbench/browser/parts/editor/breadcrumbsPicker';
+
+export class BreadcrumbObserver implements IBreadcrumbObserver {
+	declare readonly _serviceBrand: undefined;
+
+	constructor(
+		@IExplorerService private readonly explorerService: IExplorerService
+	) { }
+
+	registerTreeListeners(tree: Tree<any, any>): void {
```
Comment (irengrig) [inline src/vs/workbench/contrib/scopeTree/browser/breadcrumbObserver.ts:21]: We can unite this with similar code in the scope tree itself - this code only changes the visibility of the icon-action.
We can think how to extract it into some utility class (maybe we can also use the same style names?).

Comment (irengrig) [reply src/vs/workbench/contrib/scopeTree/browser/breadcrumbObserver.ts:21]: (let us deal with it in a follow-up CL)

File: src/vs/workbench/contrib/scopeTree/browser/breadcrumbObserver.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+			if (element) {
+				const icon = document.getElementById('breadcrumbFocusIconContainer_' + (element as IFileStat).resource.toString());
+				if (icon) {
+					icon.style.visibility = 'hidden';
+				}
+			}
+		});
+	}
+
+	renderFocusIcon(resource: URI, fileKind: FileKind, templateData: IResourceLabel): void {
```
Comment (irengrig) [inline src/vs/workbench/contrib/scopeTree/browser/breadcrumbObserver.ts:47]: Here also the onclick function will be defined, which will be delegated to the file tree - so it is completely appropriate to have this method in the observer.

File: src/vs/workbench/contrib/scopeTree/browser/breadcrumbObserver.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+	registerTreeListeners(tree: Tree<any, any>): void {
+		tree.onMouseOver(e => {
+			const element = e.element;
+			if (element) {
+				const icon = document.getElementById('breadcrumbFocusIconContainer_' + (element as IFileStat).resource.toString());
+				if (icon) {
+					icon.style.visibility = 'visible';
+					icon.onclick = () => {
+						let resource = (element as IFileStat).resource;
```
Comment (matthiasklostermann) [inline src/vs/workbench/contrib/scopeTree/browser/breadcrumbObserver.ts:29]: const

File: src/vs/workbench/contrib/scopeTree/browser/breadcrumbObserver.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+export class BreadcrumbObserver implements IBreadcrumbObserver {
+	declare readonly _serviceBrand: undefined;
+
+	constructor(
+		@IExplorerService private readonly explorerService: IExplorerService
+	) { }
+
+	registerTreeListeners(tree: Tree<any, any>): void {
+		tree.onMouseOver(e => {
+			const element = e.element;
```
Comment (matthiasklostermann) [inline src/vs/workbench/contrib/scopeTree/browser/breadcrumbObserver.ts:23]: You could already do

const element = e.element as IFileStat;

here. That will shorten the lines below a bit. The same goes for the other event listener.

**Review by matthiasklostermann — APPROVED** [review]
Just some details to look into before merging this PR.

File: src/vs/workbench/contrib/scopeTree/browser/breadcrumbObserver.ts (hunk 1 above)
Comment (bf319) [reply src/vs/workbench/contrib/scopeTree/browser/breadcrumbObserver.ts:21]: Got it! I've made a new PR for this https://github.com/irengrig/vscode/pull/43.

**Pull request merged**